### DISCUSSION TITLE
How does MediatR determine the order of notification handler?

### DISCUSSION THREAD

**Discussion by RJR-Robert-Kewei-Zhang** [LuckyPennySoftware/MediatR#779] [Q&A]
Hi @jbogard,

If I have more than one notification handlers (let say four handers subscribe the same notification), when this notification was published, how does the MediatR determine the execution order for these notification handlers?

Thanks,
Robert

**Comment by jbogard** [comment]
MediatR does not. That's the responsibility of the container.

**Comment by rangers-globecar** [comment]
Hi,
Would it be possible to decorate  handlers (and behaviors) with some priority attribute, that would be taken into account when resolving /building the pipeline?

**Reply by jbogard** [reply]
The way I'd like to solve this is in the MediatR DI package, where I would create some extension there to register the behaviors in a specific order.

**Comment by RJR-Robert-Kewei-Zhang** [comment]
@jbogard 

My notification handlers are as follow, it always start from \Handler5, looks like MediatR is using descending. Is that normal order?

Thanks

\Eventhandlers
  \Handler0
    EvenHandler.cs -- order: 6
  \Handler1
    EvenHandler.cs -- order: 5
  \Handler2
    EvenHandler.cs -- order: 4
  \Handler3
    EvenHandler.cs -- order: 3
  \Handler4
    EvenHandler.cs -- order: 2
  \Handler5
    EvenHandler.cs -- order: 1

**Reply by jbogard** [reply]
MediatR is not doing any sort of ordering, that's completely up to the underlying container. It simply asks the service provider for an enumerable of handlers. How the container decides to resolve those is separate from MediatR.
